test/missing.jl

```diff
@@ -0,0 +1,241 @@
+    @test promote_type(Any, Union{Int, Missing}) == Any
+    @test promote_type(Union{Int, Missing}, Union{Int, Missing}) == Union{Int, Missing}
+    @test promote_type(Union{Float64, Missing}, Union{String, Missing}) == Any
+    @test promote_type(Union{Float64, Missing}, Union{Int, Missing}) == Union{Float64, Missing}
+    @test_broken promote_type(Union{Void, Missing, Int}, Float64) == Any
+end
+
+@testset "comparison operators" begin
+    @test (missing == missing) === missing
+    @test (1 == missing) === missing
+    @test (missing == 1) === missing
+    @test (missing != missing) === missing
+    @test (1 != missing) === missing
+    @test (missing != 1) === missing
+    @test isequal(missing, missing)
+    @test !isequal(1, missing)
+    @test !isequal(missing, 1)
+    @test (missing < missing) === missing
+    @test (missing < 1) === missing
+    @test (1 < missing) === missing
+    @test (missing <= missing) === missing
+    @test (missing <= 1) === missing
+    @test (1 <= missing) === missing
+    @test !isless(missing, missing)
+    @test !isless(missing, 1)
+    @test isless(1, missing)
+end
+
+@testset "arithmetic operators" begin
+    arithmetic_operators = [+, -, *, /, ^, Base.div, Base.mod, Base.fld, Base.rem]
+
+    # All unary operators return missing when evaluating missing
+    for f in [!, +, -]
+        @test ismissing(f(missing))
+    end
+
+    # All arithmetic operators return missing when operating on two missing's
+    # All arithmetic operators return missing when operating on a scalar and an missing
+    # All arithmetic operators return missing when operating on an missing and a scalar
+    for f in arithmetic_operators
+        @test ismissing(f(missing, missing))
+        @test ismissing(f(1, missing))
+        @test ismissing(f(missing, 1))
+    end
+end
+
+@testset "bit operators" begin
+    bit_operators = [&, |, ⊻]
+
+    # All bit operators return missing when operating on two missing's
+    for f in bit_operators
+        @test ismissing(f(missing, missing))
+    end
+end
+
+@testset "boolean operators" begin
+    @test ismissing(missing & true)
+    @test ismissing(true & missing)
+    @test !(missing & false)
+    @test !(false & missing)
+    @test ismissing(missing | false)
+    @test ismissing(false | missing)
+    @test missing | true
+    @test true | missing
+    @test ismissing(xor(missing, true))
+    @test ismissing(xor(true, missing))
+    @test ismissing(xor(missing, false))
+    @test ismissing(xor(false, missing))
+
+    @test ismissing(missing & 1)
+    @test ismissing(1 & missing)
+    @test ismissing(missing | 1)
+    @test ismissing(1 | missing)
+    @test ismissing(xor(missing, 1))
+    @test ismissing(xor(1, missing))
+end
+
+@testset "* string concatenation" begin
+    @test ismissing("a" * missing)
+    @test ismissing(missing * "a")
+end
+
+# Emulate a unitful type such as Dates.Minute
+struct Unit
+    value::Int
+end
+Base.zero(::Type{Unit}) = Unit(0)
+Base.one(::Type{Unit}) = 1
```

Comment:
I thought it was best to not rely on the stdlib `Dates` to make these tests work. If others feel differently I can change this.